### PR TITLE
fix: IKEA VINDSTYRKA uses different DataType for measuredValue

### PR DESCRIPTION
**Pull request by sjorge**
Now that we have https://github.com/Koenkk/zigbee-herdsman/pull/1019, we can properly fix the IKEA VINDSTYRKA.

- [x] modernExtend to easily inject them by abstracting the onEvent stuff
- [x] VINDSTYRKA should use correct DataType
- [x] make SONOFF device also use new modernExtend

This is the debug log of joining a VINDSTYRKA and reconfiguring the reporting for pm25Measurement.measuredValue via the frontend.

[vindstyrka_join_and_configreport.log](https://github.com/Koenkk/zigbee-herdsman-converters/files/15137706/vindstyrka_join_and_configreport.log)

### CODE REVIEW

File: src/lib/modernExtend.ts [hunk 1]
```diff
@@ -1669,6 +1670,16 @@ export function deviceEndpoints(args: {endpoints: {[n: string]: number}, multiEn
     return result;
 }
 
+export function deviceAddCluster(clusterName: string, clusterDefinition: ClusterDefinition): ModernExtend {
```
Comment (Koenkk) [inline src/lib/modernExtend.ts:1673]:
```suggestion
export function deviceAddCustomCluster(clusterName: string, clusterDefinition: ClusterDefinition): ModernExtend {
```

**Comment by Koenkk** [comment]
Looks really nice with the modernExtend refactor!

**Comment by sjorge** [comment]
Renamed and rebased.

**Comment by sjorge** [comment]
I forgot to mention, this should fix all the vindstyrka issues either people trying to reconfigure reporting 😀

**Pull request merged**